external-api/src/v1/controllers/party-db.controller.ts

```diff
@@ -18,7 +18,7 @@ const headers = {
 export const lookup = async (req: Request, res: Response) => {
   const { partyDbCompanyRegistrationNumber: companyReg } = req.params;
 
-  if (!isValidCompaniesHouseNumber(companyReg)) {
+  if (!isValidCompanyRegistrationNumber(companyReg)) {
```

Comment:
The reason why we don't validate on `companies.controller` is because there's no need, as this same validation happens both at the very start of the flow (i.e., portal/gef ui) and once we get to MDM.

I've checked and we don't currently do this regex validation in MDM on this endpoint (which wasn't covered by this ticket), so I don't especially want to remove it here.

Reply:
The reason why we don't validate on `companies.controller` is because there's no need, as this same validation happens both at the very start of the flow (i.e., portal/gef ui) and once we get to MDM.

I've checked and we don't currently do this regex validation in MDM on this endpoint (which wasn't covered by this ticket), so I don't especially want to remove it here.